cli/core/pkg/command/config.go

```diff
@@ -216,39 +216,29 @@ var initConfigCmd = &cobra.Command{
 		if cfg.ClientOptions.CLI == nil {
 			cfg.ClientOptions.CLI = &configapi.CLIOptions{}
 		}
-		repos := cfg.ClientOptions.CLI.Repositories
-		finalRepos := []configapi.PluginRepository{}
-		for _, repo := range config.DefaultRepositories {
-			var exists bool
-			for _, r := range repos {
-				if repo.GCPPluginRepository.Name == r.GCPPluginRepository.Name {
-					finalRepos = append(finalRepos, r)
-					exists = true
-				}
-			}
-			if !exists {
-				finalRepos = append(finalRepos, repo)
-			}
+
+		serverName := ""
+		server, err := cfg.GetCurrentServer()
+		if err == nil && server != nil {
+			serverName = server.Name
 		}
-		cfg.ClientOptions.CLI.Repositories = finalRepos
 
-		err = configlib.StoreClientConfig(cfg)
+		serverPluginDescriptors, standalonePluginDescriptors, err := pluginmanager.InstalledPlugins(serverName)
 		if err != nil {
 			return err
 		}
-		// TODO: cli.ListPlugins is deprecated: Use pluginmanager.AvailablePluginsFromLocalSource or pluginmanager.AvailablePlugins instead
-		descriptors, err := cli.ListPlugins()
+
+		for _, desc := range append(serverPluginDescriptors, standalonePluginDescriptors...) {
```

Comment:
The (newly added) context-specific aspect of `config init` is not very intuitive (especially for folks who have used it before without having to think about this aspect). To properly reconstruct all the defaults in the config file, one may now have to switch contexts so that all plugins which offers default values are covered. If the user does not learn of this new behavior, he could think that a single `config init` would `fix/initialize` the config file, without realizing that it only partially initializes the file.

Would iterating through all installed plugins be an option?

If we choose to stick with the current implementation, I would still suggest
- putting a comment to note this behavior
- see if we can provide some doc and/or UX feedback to the user about the same (maybe in the help text of the config init command)

Reply:
I think iterating through all installed plugins might be overkill. Considering user might have substantial amount of plugins installed and some of them are not used anymore.

I think adding a more details in the `tanzu config init` help text and providing comment in the code should be sufficient I feel.

Reply:
I think that sounds reasonable. We should make sure any docs we produce capture the details as well. 
I suggest we tag the issue and this PR with docs-triage-needed to raise some awareness.